airbyte-integrations/connectors/source-netsuite/source_netsuite/source.py

```diff
@@ -0,0 +1,135 @@
+        )
+
+    def base_url(self, config: Mapping[str, Any]) -> str:
+        realm = config["realm"]
+        subdomain = realm.lower().replace("_", "-")
+        return f"https://{subdomain}.suitetalk.api.netsuite.com"
+
+    def get_session(self, auth: OAuth1) -> requests.Session:
+        session = requests.Session()
+        session.auth = auth
+        return session
+
+    def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        auth = self.auth(config)
+        object_types = config.get("object_types")
+        base_url = self.base_url(config)
+        session = self.get_session(auth)
+        # if record types are specified make sure they are valid
+        if object_types:
+            # ensure there are no duplicate record types as this will break Airbyte
+            duplicates = [k for k, v in Counter(object_types).items() if v > 1]
+            if duplicates:
+                return False, f'Duplicate record type: {", ".join(duplicates)}'
+            # check connectivity to all provided `object_types`
+            url = base_url + RECORD_PATH
+            for object in object_types:
+                try:
+                    response = session.get(url=url + object, params={"limit": 1})
+                    response.raise_for_status()
+                    return True, None
+                except requests.exceptions.HTTPError as e:
+                    return False, e
+        else:
+            # if `object_types` are not provided, use `Contact` stream
+            # there should be at least 1 contact available in every NetSuite account by default.
+            url = base_url + RECORD_PATH + "contact"
+            try:
+                response = session.get(url=url, params={"limit": 1})
+                response.raise_for_status()
+                return True, None
+            except requests.exceptions.HTTPError as e:
+                return False, e
+
+    def get_schemas(self, object_names: Union[List[str], str], session: requests.Session, metadata_url: str) -> Mapping[str, Any]:
+        # fetch schemas
+        if isinstance(object_names, list):
+            return {object_name: session.get(metadata_url + object_name, headers=SCHEMA_HEADERS).json() for object_name in object_names}
+        elif isinstance(object_names, str):
+            return {object_names: session.get(metadata_url + object_names, headers=SCHEMA_HEADERS).json()}
+
+    def generate_stream(
+        self,
+        session: requests.Session,
+        metadata_url: str,
+        schemas: dict,
+        object_name: str,
+        auth: OAuth1,
+        base_url: str,
+        start_datetime: str,
+        window_in_days: int,
+    ) -> Union[NetsuiteStream, IncrementalNetsuiteStream, CustomIncrementalNetsuiteStream]:
+        # try to build the stream instance dynamicaly
+        try:
+            schema = schemas[object_name]
+            schema_props = schema["properties"]
+            if schema_props:
```

Comment:
can properties be empty?

Reply:
Properties could be missing as key in schema, sometimes it happens